clikt/src/commonMain/kotlin/com/github/ajalt/clikt/parameters/options/FlagOption.kt

```diff
@@ -79,8 +80,16 @@ inline fun <OutT> OptionWithValues<Boolean, Boolean, Boolean>.convert(
 /**
  * Turn an option into a flag that counts the number of times it occurs on the command line.
  */
-fun RawOption.counted(): OptionWithValues<Int, Int, Int> {
-    return int().transformValues(0..0) { it.lastOrNull() ?: 1 }.transformAll { it.sum() }
+@JvmOverloads // TODO(5.0): remove this annotation
+fun RawOption.counted(limit: Int = 0, clamp: Boolean = true): OptionWithValues<Int, Int, Int> {
+    return int().transformValues(0..0) { it.lastOrNull() ?: 1 }.transformAll {
+        val s = it.sum()
+        when {
+            limit > 0 && clamp -> s.coerceAtMost(limit)
+            limit in 1..<s -> fail(context.localization.countedOptionExceededLimit(s, limit))
+            else -> s
+        }
```

Comment:
Plus if you use `Int.MAX_VALUE`, this simplifies to:
```suggestion
        if (!clamp && s > limit) {
            fail(context.localization.countedOptionExceededLimit(s, limit))
        }
        return s.coerceAtMost(limit)
```